parquet/src/main/java/org/apache/iceberg/parquet/ParquetBloomRowGroupFilter.java

```diff
@@ -334,5 +335,10 @@ private <T> boolean shouldRead(
           return ROWS_MIGHT_MATCH;
       }
     }
+
+    @Override
+    public <T> Boolean handleNonReference(Bound<T> term) {
+      return ROWS_MIGHT_MATCH;
```

Comment:
I think here the `NonReference` should be `it is not a column reference term`. It can be a transform term, for example `equal(year(ts), 10)`.

Reply:
@Fokko, I think at this point the columns are bound and transforms have already been removed if they can be (e.g. if they can be translated to a normal predicate on a partition field). This is the case where the transform is left over because there is no corresponding partition field. We could implement future optimizations to run the filter, but returning all rows is good enough for now.